packages/@react-spectrum/combobox/docs/ComboBox.mdx

```diff
@@ -522,8 +522,8 @@ If a description is added, the prop `slot="description"` must be used to disting
 ComboBox supports loading data asynchronously, and will display a progress circle reflecting the current load state,
 set by the `loadingState` prop. It also supports infinite scrolling to load more data on demand as the user scrolls, via the `onLoadMore` prop.
 
-This example uses the [useAsyncList](../react-stately/useAsyncList.html) hook to handle loading the data.
-See the docs for more information.
+This example uses the [useAsyncList](../react-stately/useAsyncList.html) hook to handle loading the data. See the docs for more information.
+**Note:** If you provide a `selectedKey` to the Combobox example below, it is your responsibility to update the input text when the `items` change or load.
```

Comment:
Maybe call out specifically that it's because `inputValue` is controlled as well. Also you need a newline if you want this in a separate paragraph.

```suggestion

**Note:** When both `inputValue` and `selectedKey` are controlled, and the `selectedKey` is set to an initial value before the `items` load, you must update the `inputValue` when the `items` load. This can be done by returning `filterText` from the `useAsyncList` load function.
```

Also I guess we should implement the ability to return `filterText` from the load function.

Should we show this in the example as well or would that make things too complex?

Reply:
Also hmm... we could probably do this automatically in `useAsyncList`? Like, we have the `selectedKey`, `filterText`, and `items` already. 🤔

Reply:
I've put a first stab at doing the above in https://github.com/adobe/react-spectrum/pull/1683/commits/67b8a9304b83a07f4efc07cc26c71a7fddc39ef5, is this kinda what you were envisioning?